src/collections/_documentation/development/sdk-dev/unified-api.md

```diff
@@ -217,6 +227,22 @@ the event.  For instance if an exception was captured the hint might carry the
 original exception object.  Not all SDKs are required to provide this.  The
 parameter however is reserved for this purpose.
 
+## Event Pipeline
+
+Any event that is captured by `capture_event` is processed in the following
+order and can be discarded at any of the stages, at which point no further
+processing happens.
+
+1. If the SDK is disabled or there is no active client, the event is discarded
+   right away. The client is active if it has a valid *transport* configured.
+2. The client does event sampling based on the configured sample rate.
+   Events will be discarded randomly, according to the sample rate.
+3. The scope is applied, using `apply_to_event`. The scopes *event processors*
+   are invoked in order and can possibly discard the event.
+4. The *before-send* hook is invoked, which can also discard the event.
+5. The event is passed to the configured transport. The transport can discard
+   the event in case it has no valid DSN, or due to rate limiting.
```

Comment:
Well, rate limiting is a server concern.

The more general thing to say is that "The default transport does not retry to send events that are not accepted by the server" (and that can be 4xx, 5xx responses from the server).

WDYT

Reply:
hm… at least for native, the transport eagerly discards anything in the `retry-after` window.

Reply:
the server can also discard events down the pipeline even if it returned 200, 200 just means it enqueued right